modules/preview/lib/preview.js

```diff
@@ -0,0 +1,250 @@
+import './editor.scss';
+
+import apiFetch from '@wordpress/api-fetch';
+import { __experimentalInspectorPopoverHeader as InspectorPopoverHeader } from '@wordpress/block-editor';
+import {
+	Button,
+	Dropdown,
+	ExternalLink,
+	Modal,
+	SelectControl,
+	Spinner,
+	__experimentalTruncate as Truncate,
+	Flex,
+	CheckboxControl,
+	Notice,
+} from '@wordpress/components';
+import { compose, useCopyToClipboard } from '@wordpress/compose';
+import { dispatch, withSelect } from '@wordpress/data';
+import { PluginPostStatusInfo } from '@wordpress/edit-post';
+import { store as editorStore } from '@wordpress/editor';
+import { useMemo, useRef, useState } from '@wordpress/element';
+import { __, sprintf } from '@wordpress/i18n';
+import { copySmall } from '@wordpress/icons';
+import { store as noticesStore } from '@wordpress/notices';
+import { registerPlugin } from '@wordpress/plugins';
+
+import CopyFromAsyncButton from './components/copy-from-async-button';
+
+/**
+ * Custom component to generate and copy a preview link in the post sidebar.
+ */
+const VIPWorkflowPreview = ( { status, postType, isUnsavedPost } ) => {
```

Comment:
Could you add some more detailed comments in this plugin? It'll make it easier to revisit this, down the line?

Reply:
I did an attempt at this in https://github.com/Automattic/vip-workflow-plugin/pull/19/commits/77c252587063e3ca1f48d1ddcd498db94ae080a1, but didn't have a lot of comments to add. Most of this code is pretty idiomatic for functional React components, and I'm not sure where most comments will provide context other than re-stating the code. I've already attempted to isolate non-idiomatic code like [`<CopyFromAsyncButton>`](https://github.com/Automattic/vip-workflow-plugin/blob/77c252587063e3ca1f48d1ddcd498db94ae080a1/modules/preview/lib/components/copy-from-async-button.js) in a separate well-documented component.

If you have any specific areas that you feel are lacking, please let me know!

Reply:
Added a couple more comments in https://github.com/Automattic/vip-workflow-plugin/pull/19/commits/ff5481791181c42aa4f7a85df8bea2130e5cd746.